internal/crdReplicator/reflection/reflector.go

```diff
@@ -179,16 +179,18 @@ func (r *Reflector) stopForResource(gvr schema.GroupVersionResource) error {
 
 	klog.Infof("[%v] Stopping reflection of %v", r.remoteClusterID, gvr)
 
-	// Check if any reflected object is still present in the remote cluster
-	objects, err := rs.remote.List(labels.Everything())
-	if err != nil {
-		klog.Errorf("[%v] Failed to stop reflection of %v: %v", r.remoteClusterID, gvr, err)
-		return err
-	}
+	// Check if any object is still present in the local or in the remote cluster
```

Comment:
Being the local objects owned by components other than the CRDReplicator shouldn't be them the ones that handle lifecycle for those resources?